onnxruntime/test/python/transformers/test_parity_dbrx_moe.py

```diff
@@ -0,0 +1,659 @@
+        self.num_local_experts = num_local_experts
+        self.output_router_logits = output_router_logits
+        self.router_aux_loss_coef = router_aux_loss_coef
+
+
+class DbrxExpertGLU(nn.Module):
+    def __init__(self, hidden_size: int, ffn_hidden_size: int, moe_num_experts: int, ffn_act_fn: dict, config: DBRXConfig):
+        super().__init__()
+        self.hidden_size = hidden_size
+        self.ffn_hidden_size = config.intermediate_size
+        self.moe_num_experts = config.num_local_experts
+        ffn_act_fn = {"name": config.hidden_act}
+
+        self.w1 = nn.Parameter(torch.empty(moe_num_experts, moe_num_experts * ffn_hidden_size, hidden_size))
+        self.v1 = nn.Parameter(torch.empty(moe_num_experts, moe_num_experts * ffn_hidden_size, hidden_size))
+        self.w2 = nn.Parameter(torch.empty(moe_num_experts, moe_num_experts * ffn_hidden_size, hidden_size))
+
+        act_fn_name = ffn_act_fn.get("name", "silu")
+        self.activation_fn = ACT2FN[act_fn_name]
+
+    def forward(
+        self, x: torch.Tensor, expert_w1: torch.Tensor, expert_v1: torch.Tensor, expert_w2: torch.Tensor
+    ) -> torch.Tensor:
+        gate_proj = x.matmul(expert_w1.t())
+        up_proj = x.matmul(expert_v1.t())
+        gate_proj = self.activation_fn(gate_proj)
+        intermediate_states = gate_proj * up_proj
+        down_proj = intermediate_states.matmul(expert_w2)
+        return down_proj
+
+
+class DbrxExperts(nn.Module):
+    def __init__(self, hidden_size: int, ffn_hidden_size: int, moe_num_experts: int, ffn_act_fn: dict, batch_size: int, sequence_length: int, config: DBRXConfig):
+        super().__init__()
+        self.moe_num_experts = config.num_local_experts
+        self.config = DBRXConfig()
+        self.hidden_size = hidden_size
+        self.ffn_hidden_size = config.intermediate_size
+        self.moe_top_k = config.num_experts_per_tok
+        self.mlp = DbrxExpertGLU(
+                    hidden_size=hidden_size,
+                    ffn_hidden_size=ffn_hidden_size,
+                    moe_num_experts=moe_num_experts,
+                    ffn_act_fn=config.hidden_act,
+                    config=config
+                    )
+
+        w1_list = []
+        v1_list = []
+        w2_list = []
+        for i in range(self.moe_num_experts):
+            w1_list.append(self.mlp.w1[i])
+            v1_list.append(self.mlp.v1[i])
+            w2_list.append(self.mlp.w2[i])
+        self.moe_experts_weight1 = torch.stack(w1_list, dim=0)
+        self.moe_experts_weight2 = torch.stack(v1_list, dim=0)
+        self.moe_experts_weight3 = torch.stack(w2_list, dim=0)
+        self.batch_size = batch_size
+        self.sequence_length = sequence_length
+        self.moe_onnx_graph = create_moe_onnx_graph(
+            self.batch_size * self.sequence_length,
+            self.moe_num_experts,
+            self.hidden_size,
+            self.ffn_hidden_size,
+            self.moe_experts_weight1,
```

Comment:
pass self.mlp.w1/w2/v1 directly since they are defined with shape [num_experts, ...]
this is the part that's different from mixtral
you probably need to transpose one of them to make it align with ORT format